File: MPP-lab-project/src/Domain/SwimmerRace.cs
using System;

namespace MPP_lab_project.Domain;

[Serializable()]
public class SwimmerRace
{
    public int ID { get; set; }
    public Swimmer Swimmer { get; }
    public Race Race { get;  }

    public SwimmerRace(Swimmer swimmer, Race race)
    {
        Swimmer = swimmer;
        Race = race;
    }

    public override string ToString()
    {
        return "SwimmerRace{" +
               "ID=" + ID +
               ", swimmer=" + Swimmer +
               ", race=" + Race +
               '}';
    }
}

File: swimming-contest-software/src/main/java/swimmingApp/domain/SwimmerRace.java
package swimmingApp.domain;

import java.io.Serializable;

public class SwimmerRace implements Identifiable<Integer>, Serializable {
    private Integer ID;
    private final Swimmer swimmer;
    private final Race race;

    public SwimmerRace(Swimmer swimmer, Race race) {
        this.swimmer = swimmer;
        this.race = race;
    }

    @Override
    public Integer getID() {
        return ID;
    }

    @Override
    public void setID(Integer id) {
        this.ID = id;
    }

    public Swimmer getSwimmer() {
        return swimmer;
    }

    public Race getRace() {
        return race;
    }

    @Override
    public String toString() {
        return "SwimmerRace{" +
                "ID=" + ID +
                ", swimmer=" + swimmer +
                ", race=" + race +
                '}';
    }
}
